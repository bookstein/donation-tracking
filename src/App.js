import React, { Component } from 'react'
import './App.css'
import ProfileContainer from './containers/profileContainer'

class App extends Component {
  render() {
    return <ProfileContainer />
  }
}

export default App
