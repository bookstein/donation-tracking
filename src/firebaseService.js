import fb from './firebase'

export const listenForUpdates = (refName, cb) => {
  let myRef = fb
    .database()
    .ref(refName)
    .orderByKey()

  myRef.on('child_added', cb)
  myRef.on('child_removed', cb)
}

export const pushToDatabase = (refName, data) => {
  const myRef = fb
    .database()
    .ref(refName)
    .push()
  myRef.set(data)
  // for some reason I can only use `set`
  // `push` seems to append to the last item in the collection
  console.log('view your new data at ', myRef.toString())
}

export const removeFromDatabase = (refName, key) => {
  const myRef = fb.database().ref(refName)
  myRef
    .child(key)
    .remove()
    .then(function() {
      console.log('Remove succeeded.')
    })
    .catch(function(error) {
      console.log('Remove failed: ' + error.message)
    })
}
