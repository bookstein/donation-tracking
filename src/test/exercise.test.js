import React from 'react'
import ReactDOM from 'react-dom'
import App from '../App'
import Exercise from '../components/exercise'

// get testing tools
import renderer from 'react-test-renderer'
import { shallow, mount } from 'enzyme'
import { removeFromDatabase } from '../firebaseService'

const removeExercise = jest.fn()
const PROPS = {
  removalKey: 'testKey',
  exerciseTag: 'testTag',
  exerciseName: 'testName',
  removeExercise,
}

// happy path: component renders (jest snapshot)
it('renders without crashing', () => {
  const component = renderer.create(<Exercise {...PROPS} />)
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

// simulate user behavior
// click X
it('calls the `remove` callback on click', () => {
  const component = shallow(<Exercise {...PROPS} />)
  component.find('.Exercise__X').simulate('click')
  expect(removeExercise).toHaveBeenCalled()
})
