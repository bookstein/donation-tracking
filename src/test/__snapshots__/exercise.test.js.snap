// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="Exercise"
>
  <div>
    <h1>
      testName
    </h1>
    <p
      className="Exercise__Tags"
    >
      Tags:
      <span
        className="Exercise__Tag"
      >
        testTag
      </span>
    </p>
    <div
      className="Exercise__Completions"
    >
      <h3>
        Recent completions:
      </h3>
      <p>
        195 lbs 5 reps
      </p>
      <p>
        185 lbs 5 reps
      </p>
      <p>
        185 lbs 5 reps
      </p>
    </div>
  </div>
  <div
    className="Exercise__X"
    onClick={[Function]}
  >
    X
  </div>
</div>
`;
